### PR TITLE
Use Tina styles for NumberInput component

### PR DESCRIPTION
**Pull request by PaulBunker**
the purpose of this pull request is for weibenfalk's number input feature to use the same styling as text input field.

### CODE REVIEW

**Comment by PaulBunker** [comment]
I could get the numberfield working if I added it in gatsby-node
`
const NumberInput = require("@tinacms/fields").NumberInput
exports.onClientEntry = () => {
  window.tinacms.fields.add({
    name: "number",
    Component: NumberInput,
  })
}`

However the plugin wasn't registering 
`Unrecognized field type`

It looks like the interface is being imported empty

`(alias) const BaseNumberInput: React.FC<NumberProps>
import BaseNumberInput
Type '{}' is missing the following properties from type 'NumberProps': name, input, field, stepsts(2739)
Peek Problem
No quick fixes available`

I'm not sure what to do next to try and fix it.

File: packages/tinacms/src/plugins/fields/NumberFieldPlugin.tsx [hunk 1]
```diff
@@ -16,14 +16,18 @@ limitations under the License.
...
+import * as React from 'react'
 import { wrapFieldsWithMeta } from './wrapFieldWithMeta'
-import { NumberInput } from '@tinacms/fields'
-import { parse } from './numberFormat';
+import { NumberInput as BaseNumberInput, NumberProps } from '@tinacms/fields'
+import { parse } from './numberFormat'
 
-export const NumberField = wrapFieldsWithMeta(NumberInput)
+export const NumberField = wrapFieldsWithMeta<
+  { placeholder: string },
```
Comment (ncphillips) [inline packages/tinacms/src/plugins/fields/NumberFieldPlugin.tsx]:
```suggestion
```

File: packages/tinacms/src/plugins/fields/NumberFieldPlugin.tsx [hunk 2]
```diff
@@ -16,14 +16,18 @@ limitations under the License.
...
+  NumberProps
+>(() => <BaseNumberInput />)
```
Comment (ncphillips) [inline packages/tinacms/src/plugins/fields/NumberFieldPlugin.tsx]: Nothing is being passed to the BaseNumberInput. That's what's causing the errors.
```suggestion
>(BaseNumberInput)
```

**Comment by ncphillips** [comment]
oops just linting errors in there now

**Comment by ncphillips** [comment]
@all-contributors please add @PaulBunker for code

**Pull request merged**